block-storage/src/main/scala/coop/rchain/blockstorage/LMDBBlockStore.scala

```diff
@@ -0,0 +1,130 @@
+package coop.rchain.blockstorage
+
+import cats._
+import cats.effect.{Bracket, ExitCase}
+import cats.implicits._
+import cats.mtl.MonadState
+import com.google.protobuf.ByteString
+import coop.rchain.blockstorage.BlockStore.BlockHash
+import coop.rchain.casper.protocol.BlockMessage
+import coop.rchain.metrics.Metrics
+
+import scala.language.higherKinds
+import scala.collection.JavaConverters._
+
+import java.nio.ByteBuffer
+import java.nio.file.Path
+import org.lmdbjava.DbiFlags.MDB_CREATE
+import org.lmdbjava._
+
+class LMDBBlockStore[F[_]] private (val env: Env[ByteBuffer], path: Path, blocks: Dbi[ByteBuffer])(
+    implicit
+    bracketF: Bracket[F, Exception],
```

Comment:
Since `Sync[F[_]] extends Bracket[F, Throwable]` I think you can summon `Sync[F]` and then use `Sync[F].delay(…)` where @henrytill noted.